src/simulator.zig

```diff
@@ -593,12 +593,16 @@ pub const Simulator = struct {
     pub fn core_missing_prepare(simulator: *const Simulator) ?vsr.Header.Prepare {
         assert(simulator.core.count() > 0);
 
+        // Don't check for missing uncommitted ops (since the StateChecker does not record them).
+        // There may be uncommitted ops due to pulses/upgrades sent during liveness mode.
+        const commit_max = simulator.cluster.state_checker.commits.items.len - 1;
```

Comment:
:thinking: so this fix will make us not crash in this function, but it'll also make this function return `null` if the uncommited prepare is indeed uncommited duo to too many faults in the core. 

So we'll get a liveness simualtor failure rather than a crash one?

Reply:
In this case it "fails" with `no liveness, op=123 is not available in core`. (op=124 is the uncommitted one.)